File: frontend/src/api/education.ts
import { http } from '@/utils/request'

/**
 * 教育经历接口类型定义
 */
export interface Education {
  id?: number
  userId?: number
  schoolName: string
  major: string
  degree: string // 本科/硕士/博士/专科
  startDate: string // YYYY-MM-DD
  endDate?: string // YYYY-MM-DD, 空表示至今
  description?: string
  createTime?: string
  updateTime?: string
}

/**
 * 根据用户ID获取教育经历列表
 */
export function getEducationByUserIdApi(userId: number) {
  return http.get<Education[]>(`/education/user/${userId}`)
}

/**
 * 根据ID获取教育经历详情
 */
export function getEducationByIdApi(id: number) {
  return http.get<Education>(`/education/${id}`)
}

/**
 * 创建教育经历
 */
export function createEducationApi(data: Education) {
  return http.post<void>('/education', data)
}

/**
 * 更新教育经历
 */
export function updateEducationApi(data: Education) {
  return http.put<void>('/education', data)
}

/**
 * 删除教育经历
 */
export function deleteEducationApi(id: number) {
  return http.delete<void>(`/education/${id}`)
}

File: backend/src/main/java/com/teaching/employment/entity/Education.java
package com.teaching.employment.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 教育经历实体类
 *
 * @author Teaching Employment Platform Team
 * @since 2026-01-18
 */
@Data
@TableName("t_education")
@ApiModel(description = "教育经历信息")
public class Education implements Serializable {

    private static final long serialVersionUID = 1L;

    @ApiModelProperty(value = "主键ID")
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @ApiModelProperty(value = "用户ID")
    private Long userId;

    @ApiModelProperty(value = "学校名称")
    private String schoolName;

    @ApiModelProperty(value = "专业")
    private String major;

    @ApiModelProperty(value = "学历(本科/硕士/博士/专科)")
    private String degree;

    @ApiModelProperty(value = "开始时间")
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate startDate;

    @ApiModelProperty(value = "结束时间(空表示至今)")
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate endDate;

    @ApiModelProperty(value = "描述/备注")
    private String description;

    @ApiModelProperty(value = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @ApiModelProperty(value = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}
